### ISSUE TITLE
Custom timeseries sample weighting

### ISSUE THREAD

**Issue by moerfii** [unit8co/darts#2030]
**Is your feature request related to a current problem? Please describe.**
Currently There is no possibility to apply custom weighting to timeseries rows.

**Describe proposed solution**
Make use of "sample_weight" parameter in model.fit() functions (https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeRegressor.html#sklearn.tree.DecisionTreeRegressor.fit).
Darts must include new column "sample_weight" to provide to .fit() functions.
Possibly provide an interface: user provides argument ["linear", "exponential", "sigmoid".. etc.], darts includes a new column of sample_weight (first initialised linearly: range from 0 to 1 in N steps (N being the set equal to the lag defined by the user). Darts applies any mathematical transformation to the new column. (Image Linear --> exponential)

<img src="https://github.com/unit8co/darts/assets/55541119/f445d112-87dc-4dab-8783-202710e50761" width="300" height="300">
<img src="https://github.com/unit8co/darts/assets/55541119/039512cd-c03a-45f0-b93e-593d61b791e4" width="300" height="300">

**Comment by madtoinou** [comment]
Hi @moerfii,

Thanks you for this detailed and clear feature request, it's already tracked by #1175.
